src/sema/variables.rs

```diff
@@ -233,9 +235,22 @@ pub fn variable_decl<'a>(
 
                 visibility = Some(v.clone());
             }
+            pt::VariableAttribute::StorageType(s) => {
+                storage_type = Some(s.clone());
```

Comment:
Here you don't check for duplicate entries. For example:

```
contract c {
    int temporary persistent v;
}
```

```

Reply:
This confused me for a second, but I assume you're suggesting that we add a check to prevent declaring two storage type modifiers for the same variable in a contract. However, in Soroban, each storage 'map' (e.g., temporary, persistent, instance) operates independently of the others. It's possible to store different (or even the same) values for the same key across different storage types. For example, the key 123_u32 might hold the value 100_u32 in temporary storage, while in persistent storage, the same key could store abcd or 100_u32.

So, if both modifiers are handled correctly, Soroban would technically allow this. That said, while it's not typically recommended—since it consumes more storage—there could be edge cases where this pattern is useful, perhaps as a way to initialize data across namespaces. In those cases, such declarations might be valid.

Reply:
I think @seanyoung meant this case:
`uint64 public temporary instance persistent sesa = 1;`